sotodlib/tod_ops/sss.py

```diff
@@ -0,0 +1,216 @@
+        Flag indicating whether to exclude flagged samples when binning the signal.
+        Default is no mask applied.
+    method: str
+        The method to use for SSS modeling. Options are 'interpolate' and 'fit'. 
+        In 'interpolate', binned signal is used directly.
+        In 'fitting', fitting is applied to the binned signal.
+        Defaults to 'interpolate'.
+    nmodes: integer, optinal
+        The number of Legendre modes to use for SSS when method is 'fit'. Required when method is 'fit'.
+    merge_stats: boolean, optional
+        Boolean flag indicating whether to merge the SSS statistics with aman. Defaults to True.
+    sss_stats_name: string, optional
+        The name to assign to the merged SSS statistics. Defaults to 'sss_stats'.
+    merge_model: boolean, optional
+        Boolean flag indicating whether to merge the SSS model with the aman. Defaults to True.
+    sss_model_name: string, optional
+        The name to assign to the merged SSS model. Defaults to 'sss_model'.
+
+    Returns
+    -------
+        tuple: A tuple containing the SSS statistics and the SSS model.
+            - sss_stats: An instance of core.AxisManager containing the SSS statistics.
+            - model_sig_tod: A numpy array representing the SSS model as a function of time.
+    """
+    if signal is None:
+        signal = aman.signal
+    if az is None:
+        az = aman.boresight.az
+        
+    # do binning
+    binning_dict = bin_by_az(aman, signal=signal, az=az, bin_range=bin_range, bins=bins, flags=flags)
+    bin_centers = binning_dict['bin_centers']
+    binned_signal = binning_dict['binned_signal']
+    binned_signal_sigma = binning_dict['binned_signal_sigma']
+    uniform_binned_signal_sigma = np.nanmedian(binned_signal_sigma, axis=-1)
+    
+    sss_stats = core.AxisManager(aman.dets)
+    sss_stats.wrap('binned_az', bin_centers, [(0, core.IndexAxis('bin_samps', count=bins))])
+    sss_stats.wrap('binned_signal', binned_signal, [(0, 'dets'), (1, 'bin_samps')])
+    sss_stats.wrap('binned_signal_sigma', binned_signal_sigma, [(0, 'dets'), (1, 'bin_samps')])
+    sss_stats.wrap('uniform_binned_signal_sigma', uniform_binned_signal_sigma, [(0, 'dets')])
+    
+    if method == 'fit':
+        sss_stats, model_sig_tod = fit_sss(az=az, sss_stats=sss_stats, nmodes=nmodes, fit_range=bin_range)
+        
+        
+    if method == 'interpolate':
+        f_template = interp1d(bin_centers, binned_signal, fill_value='extrapolate')
+        model_sig_tod = f_template(aman.boresight.az)
+    
+    if merge_stats:
+        aman.wrap(sss_stats_name, sss_stats)
+    if merge_model:
+        aman.wrap(sss_model_name, model_sig_tod, [(0, 'dets'), (1, 'samps')])
+    
+    return sss_stats, model_sig_tod
+
+def subtract_sss(aman, signal=None, sss_template=None,
+                 subtract_name='sss_remove'):
+    """
+    Subtract the scan synchronous signal (SSS) template from the
+    signal in the given axis manager.
+
```

Comment:
Clarify that this does not modify the signal array directly, but rather produces and returns a new array that results from subtracting the template from the provided signal.

I think it would be more useful to have a function (or an option in this function) that modifies the signal _in place_.  Remember that signal arrays are often big, and it is expensive to create and store copies of them.

Reply:
I think this style of function, get-model and subtract-model, is defined from preprocessing. It is used  also used in hwp synchronous signal derivation/subtraction. Maybe there is a way to satisfy both.

Reply:
I have added `subtract_in_place` optinon in get_azss. So user can just subtract sss with `subtract_in_place`=True and `merge_model`=False without increasing the size of data.